python/ciqueue/pytest.py

```diff
@@ -114,12 +110,25 @@ def pytest_runtest_makereport(self, item, call):
                 self.terminalwriter.write(' TIMED OUT ', green=True)
 
 
-@pytest.hookimpl(trylast=True)
-def pytest_collection_modifyitems(session, config, items):
-    """This function hooks into pytest's list of tests to run, and replaces
-    those `items` with a redis test queue iterator."""
-    tests_index = ItemIndex(items)
+@pytest.hookimpl(tryfirst=True)
+def pytest_runtestloop(session):
```

Comment:
90% copy-pasta'd from [here](https://github.com/pytest-dev/pytest/blob/master/_pytest/main.py#L158-L172) but with our ci queue inserted after we check for collection errors.